buildSrc/src/main/groovy/org/robolectric/gradle/DeployedRoboJavaModulePlugin.groovy

```diff
@@ -9,6 +9,7 @@ class DeployedRoboJavaModulePlugin implements Plugin<Project> {
     Closure doApply = {
         project.apply plugin: "signing"
         project.apply plugin: "maven-publish"
+        project.apply plugin: "org.robolectric.gradle.RoboJavaModulePlugin"
```

Comment:
Not sure why it is necessary now, but it's because `DeployedRoboJavaModulePlugin` uses `mavenArtifactName`:
https://github.com/robolectric/robolectric/blob/dce3d2eca19869b254722478af0d4fce04f464e7/buildSrc/src/main/groovy/org/robolectric/gradle/DeployedRoboJavaModulePlugin.groovy#L34

Which is defined in `RoboJavaModulePlugin `:
https://github.com/robolectric/robolectric/blob/dce3d2eca19869b254722478af0d4fce04f464e7/buildSrc/src/main/groovy/org/robolectric/gradle/RoboJavaModulePlugin.groovy#L41

Reply:
Having a second look at my changes, I think it's because I sorted the plugins alphabetically:
<img width="1151" alt="Screenshot 2024-06-29 at 14 03 20" src="https://github.com/robolectric/robolectric/assets/1009664/b0aea1c6-7b6e-4897-9eae-ed20af7e970c">

Before the `RoboJavaModulePlugin` was applied before `DeployedRoboJavaModulePlugin`, and now it's the opposite.

So if we want to, we could make it work without that extra line. I just have to revert the ordering change.

Reply:
That seems to work, I've done the change: https://github.com/robolectric/robolectric/compare/3df0ccf8f9904ebc3ce812636ba33b90d37958ad..66800c1c778724868901af90e3459c2f05faf54a